web_poet/serialization/api.py

```diff
@@ -135,7 +136,19 @@ def serialize(deps: Iterable[Any]) -> SerializedData:
         cls = dep.__class__
         if is_injectable(cls):
             raise ValueError(f"Injectable type {cls} passed to serialize()")
-        result[_get_name_for_class(cls)] = serialize_leaf(dep)
+        if cls is AnnotatedResult:
+            key = f"AnnotatedResult {_get_name_for_class(dep.result.__class__)}"
```

Comment:
I needed a new separator and used a space. Suggestions for a better one a welcome.

Reply:
What does it affect? Names of files on disk?

Reply:
Yes.